demo-project/src/demo_project/settings.py

```diff
@@ -4,14 +4,49 @@
 
 from pathlib import Path
 
+from traitlets import default
+
 # Define where to store data from a KedroSession. Defaults to BaseSessionStore.
 # from kedro.framework.session.store import ShelveStore
 from kedro_viz.integrations.kedro.sqlite_store import SQLiteStore
 
+from kedro.framework.hooks import hook_impl
+from collections import defaultdict
+
+
+class DatasetStatsHook:
+    def __init__(self):
+        self._stats = defaultdict(dict)
+
+    @hook_impl
+    def after_context_created(self, context):
+        self._catalog = context.catalog
+
+    @hook_impl
+    def after_dataset_loaded(self, dataset_name, data):
+        import pandas as pd
```

Comment:
Should this be in a `try/except` if the user doesn't have pandas installed?

Reply:
This is written in 20 minutes gap between meeting, definitely not ready to be merged 😂